include/ginkgo/core/base/polymorphic_object.hpp

```diff
@@ -186,6 +186,27 @@ class PolymorphicObject : public log::EnableLogging<PolymorphicObject> {
         return copied;
     }
 
+    /**
+     * Moves another object into this object.
+     *
+     * This is the polymorphic equivalent of the move assignment operator.
+     *
+     * @see move_from_impl(PolymorphicObject *)
+     *
+     * @param other  the object to move from
+     *
+     * @return this
+     */
+    PolymorphicObject* move_from(PolymorphicObject* other)
+    {
+        this->template log<log::Logger::polymorphic_object_copy_started>(
```

Comment:
Should we still be using the `*_copy_started` masks for move as well ? As it is not a deep copy, I think it might make sense to add different masks like `*_move_started` and *_move_completed`.